Neos.Flow/Classes/Persistence/Doctrine/DataTypes/ObjectArray.php

```diff
@@ -20,113 +25,74 @@
-     * can be used when binding parameters to prepared statements.
-     *
-     * @return integer
+     * Use LARGE_OBJECT instead of STRING
      */
-    public function getBindingType()
+    public function getBindingType(): int
     {
-        return \PDO::PARAM_LOB;
+        return ParameterType::LARGE_OBJECT;
     }
 
     /**
-     * Converts a value from its database representation to its PHP representation
-     * of this type.
-     *
-     * @param mixed $value The value to convert.
-     * @param AbstractPlatform $platform The currently used database platform.
-     * @return array The PHP representation of the value.
+     * @throws ConversionException
      */
-    public function convertToPHPValue($value, AbstractPlatform $platform)
+    public function convertToPHPValue($value, AbstractPlatform $platform): array
     {
         $this->initializeDependencies();
 
-        switch ($platform->getName()) {
-            case 'postgresql':
-                $value = (is_resource($value)) ? stream_get_contents($value) : $value;
-                $array = parent::convertToPHPValue(hex2bin($value), $platform);
-            break;
-            default:
-                $array = parent::convertToPHPValue($value, $platform);
+        if ($platform instanceof PostgreSQL94Platform) {
+            $value = hex2bin((is_resource($value)) ? stream_get_contents($value) : $value);
         }
-        $this->decodeObjectReferences($array);
 
-        return $array;
+        $value = parent::convertToPHPValue($value, $platform);
+        $this->decodeObjectReferences($value);
+
+        return $value;
     }
 
     /**
      * Converts a value from its PHP representation to its database representation
      * of this type.
-     *
-     * @param mixed $array The value to convert.
-     * @param AbstractPlatform $platform The currently used database platform.
-     * @return mixed The database representation of the value.
      */
-    public function convertToDatabaseValue($array, AbstractPlatform $platform)
+    public function convertToDatabaseValue($value, AbstractPlatform $platform): mixed
     {
-        if (!is_array($array)) {
-            throw new \InvalidArgumentException(sprintf('The ObjectArray type only converts arrays, %s given', gettype($array)), 1569945649);
+        if (!is_array($value)) {
+            throw new \InvalidArgumentException(sprintf('The ObjectArray type only converts arrays, %s given', gettype($value)), 1569945649);
         }
 
         $this->initializeDependencies();
+        $this->encodeObjectReferences($value);
 
-        $this->encodeObjectReferences($array);
-
-        switch ($platform->getName()) {
-            case 'postgresql':
-                return bin2hex(parent::convertToDatabaseValue($array, $platform));
-            default:
-                return parent::convertToDatabaseValue($array, $platform);
+        if ($platform instanceof PostgreSQL94Platform) {
```

Comment:
i trust that `$platform instanceof PostgreSQL94Platform` is the same as `$platform->getName() == "postgresql"` ?

Reply:
It's a match, yes. But it's even better, as it more closely checks what we expect.